src/coreclr/jit/codegenloongarch64.cpp

```diff
@@ -0,0 +1,9541 @@
+            }
+            else if (tree->OperIs(GT_GE))
+            {
+                emit->emitIns_R_R_I(cmpSize == EA_4BYTE ? INS_fcmp_cle_s : INS_fcmp_cle_d, cmpSize, op2->GetRegNum(),
+                                    op1->GetRegNum(), 1 /*cc*/);
+            }
+        }
+
+        emit->emitIns_R_R(INS_mov, EA_PTRSIZE, targetReg, REG_R0);
+        emit->emitIns_R_I(INS_movcf2gr, EA_PTRSIZE, targetReg, 1 /*cc*/);
+    }
+    else if (op1->isContainedIntOrIImmed() && op2->isContainedIntOrIImmed())
+    {
+        ssize_t imm1 = op1->AsIntCon()->gtIconVal;
+        ssize_t imm2 = op2->AsIntCon()->gtIconVal;
+
+        assert(tree->OperIs(GT_LT, GT_LE, GT_EQ, GT_NE, GT_GT, GT_GE));
+
+        bool        IsUnsigned = (tree->gtFlags & GTF_UNSIGNED) != 0;
+        instruction ins        = INS_beqz;
+
+        switch (cmpSize)
+        {
+            case EA_4BYTE:
+            {
+                imm1 = static_cast<int32_t>(imm1);
+                imm2 = static_cast<int32_t>(imm2);
+            }
+            break;
+            case EA_8BYTE:
+                break;
+            case EA_1BYTE:
+            {
+                imm1 = static_cast<int8_t>(imm1);
+                imm2 = static_cast<int8_t>(imm2);
+            }
+            break;
+            // case EA_2BYTE:
+            //    imm = static_cast<uint16_t>(imm);
+            //    break;
+            default:
+                assert(!"Unexpected type in jumpCompare.");
+        }
+
+        switch (tree->OperGet())
+        {
+            case GT_LT:
+                if (((!IsUnsigned) && (imm1 < imm2)) || ((IsUnsigned) && ((unsigned)imm1 < (unsigned)imm2)))
+                {
+                    ins = INS_b;
+                }
+                break;
+            case GT_LE:
+                if (((!IsUnsigned) && (imm1 <= imm2)) || ((IsUnsigned) && ((unsigned)imm1 <= (unsigned)imm2)))
+                {
+                    ins = INS_b;
+                }
+                break;
+            case GT_EQ:
+                if (imm1 == imm2)
+                {
+                    ins = INS_b;
+                }
+                break;
+            case GT_NE:
+                if (imm1 != imm2)
+                {
+                    ins = INS_b;
+                }
+                break;
+            case GT_GT:
+                if (((!IsUnsigned) && (imm1 > imm2)) || ((IsUnsigned) && ((unsigned)imm1 > (unsigned)imm2)))
+                {
+                    ins = INS_b;
+                }
+                break;
+            case GT_GE:
+                if (((!IsUnsigned) && (imm1 >= imm2)) || ((IsUnsigned) && ((unsigned)imm1 >= (unsigned)imm2)))
+                {
+                    ins = INS_b;
+                }
+                break;
+            default:
+                break;
+        }
+
+        assert(ins != INS_invalid);
+        jtree->gtOp2 = (GenTree*)REG_SP;
+        jtree->SetRegNum((regNumber)ins);
+    }
```

Comment:
```suggestion
```
Doubly contained immediates no longer exist.

Reply:
Ok, I will delete it.
Thanks.